devito/data/decomposition.py

```diff
@@ -270,6 +286,92 @@ def convert_index(self, *args, rel=True):
         else:
             raise TypeError("Expected 1 or 2 arguments, found %d" % len(args))
 
+    def convert_index_global(self, *args, rel=True):
```

Comment:
I think it's worth renaming the `convert_index*` methods into something like...
```
convert_index_glb_to_loc
```
and 
```
convert_index_loc_to_glb
```

or maybe something else...as long as the symmetry between the two is clear

Reply:
I've gone with `_convert_index_loc` and `_convert_index_glb` for now?